contracts/stake-cw20-external-rewards/src/contract.rs

```diff
@@ -716,6 +728,55 @@ mod tests {
             .unwrap();
     }
 
+    #[test]
+    fn test_zero_rewards_duration() {
+        let mut app = mock_app();
+        let admin = Addr::unchecked(OWNER);
+        app.borrow_mut().update_block(|b| b.height = 0);
+        let initial_balances = vec![
+            Cw20Coin {
+                address: ADDR1.to_string(),
+                amount: Uint128::new(100),
+            },
+            Cw20Coin {
+                address: ADDR2.to_string(),
+                amount: Uint128::new(50),
+            },
+            Cw20Coin {
+                address: ADDR3.to_string(),
+                amount: Uint128::new(50),
+            },
+        ];
```

Comment:
Just a small nitpick, to reduce a bit the verbosity of the test (we don't need any addresses to verify the instantiate validation), we could use an empty vector here, especially if [this](https://github.com/DA0-DA0/dao-contracts/pull/349/files#r894816126) is done.

I had sent a similar PR for this issue by mistake, here's how I've written the test: https://github.com/DA0-DA0/dao-contracts/pull/355/files

Reply:
Wow. Thanks for such a close review. Great point!